presto-parquet/src/main/java/io/prestosql/parquet/predicate/TupleDomainParquetPredicate.java

```diff
@@ -389,4 +502,353 @@ private static ParquetCorruptionException corruptionException(String column, Par
+                    Binary max = (Binary) statistic.getMax();
+                    return canDropCanWithRangeStats(new ParquetStringStatistics((Slices.wrappedBuffer(min.getBytes())), Slices.wrappedBuffer(max.getBytes())));
+                }
+                //TODO: Add other types
+            }
+            return false;
+        }
+
+        @Override
+        public boolean inverseCanDrop(org.apache.parquet.filter2.predicate.Statistics<T> statistics)
+        {
+            // !canDrop() cannot be used because it might not be correct. To be safe, we just keep the record by returning false.
+            // Since we don't use LogicalNotUserDefined, this method is not called.
+            return false;
+        }
+
+        private boolean canDropCanWithRangeStats(ParquetRangeStatistics parquetStatistics)
+        {
+            //TODO: hasNullValue is set true. Would it generate false negative?
+            Domain domain = createDomain(columnDomain.getType(), true, parquetStatistics);
+            if (columnDomain.intersect(domain).isNone()) {
+                return true;
+            }
+            return false;
+        }
+    }
+
+    class ColumnIndexValueConverter
+    {
+        private final Map<String, Function<Object, Object>> ciConversions;
+
+        private ColumnIndexValueConverter(List<RichColumnDescriptor> columns)
+        {
+            this.ciConversions = new HashMap<>();
+            for (RichColumnDescriptor column : columns) {
+                ciConversions.put(column.getPrimitiveType().getName(), getColumnIndexConversions(column.getPrimitiveType()));
+            }
+        }
+
+        public List<Long> getMinValuesAsLong(Type type, ColumnIndex columnIndex, String column)
+        {
+            int pageCount = columnIndex.getMinValues().size();
+            List<ByteBuffer> minValues = columnIndex.getMinValues();
+            List<Long> mins = new ArrayList<>();
+            for (int i = 0; i < pageCount; i++) {
+                if (TINYINT.equals(type) || SMALLINT.equals(type) || INTEGER.equals(type)) {
+                    int minValue = converter.convert(minValues.get(i), column);
+                    mins.add((long) minValue);
+                }
+                else if (BIGINT.equals(type)) {
+                    long minValue = converter.convert(minValues.get(i), column);
+                    mins.add(minValue);
+                }
+                else if (REAL.equals(type)) {
+                    int minValue = floatToRawIntBits(converter.convert(minValues.get(i), column));
+                    mins.add((long) minValue);
+                }
+            }
+            return mins;
+        }
+
+        public List<Long> getMaxValuesAsLong(Type type, ColumnIndex columnIndex, String column)
+        {
+            int pageCount = columnIndex.getMaxValues().size();
+            List<ByteBuffer> maxValues = columnIndex.getMaxValues();
+            List<Long> maxs = new ArrayList<>();
```

Comment:
I think `maxes` is a better name.

I can't help but read `maxs` as `maxis` or `max s`.

This shows up in a few other places that I might have missed as well.